lib/pantry.rb

```diff
@@ -0,0 +1,30 @@
+class Pantry
+
+    attr_reader :stock
+
+    def initialize
+        @stock = Hash.new(0)
+    end
+    
+    def restock(ingredient, qty)
+        @stock[ingredient] += qty
+    end
+
+    def stock_check(ingredient = "")
+        if ingredient == ""
+            if @stock.values.inject(:+) == nil
+                0
+            else 
+                @stock.values.inject(:+)
+            end
+        else
+            @stock[ingredient]
+        end
+    end
+
+    def enough_ingredients_for?(recipe)
+        @stock.all? do |ingredient|
```

Comment:
Remember when we iterate over a hash, we get two block variables. Great use of .all? though.